### PR TITLE
Proposed change to remove whitespace for issue #39370

### LINKED ISSUE #39370: Same result but different outcome based on the 'code style'
**Describe your problem and how to reproduce it:**
In this challenge(screenshot below):
image

If I answer the question as I typed in the code editor, it won't pass... and get error like this:
image

But if I change the answered code into a different code style, it will pass....
image

image

**Add a Link to the page with the problem:**
https://www.freecodecamp.org/learn/front-end-libraries/react/render-state-in-the-user-interface

**Tell us about your browser and operating system:**
* Browser Name: Chrome 
* Browser Version: Version 84.0.4147.105 (Official Build) (64-bit)
* Operating System: macOS 10.15.6

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**
Check above~

### PR DESCRIPTION
**Pull request by CheongJY**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Closes #39370

Description:
Issue #39370: whitespace fails the 3rd test with test comparing to a no whitespace example.
Proposed: remove all whitespace before comparing it with the no whitespace example

### CODE REVIEW

**Review by ShaunSHamilton — DISMISSED** [review]
Just a small change to capture all kinds of space: `\r\n\t\f\v`

**Comment by Sky020** [comment]
The linter seems to think the testline should end with a `;`. Which, in my understanding, is unnecessary for testStrings enclosed in quotes? @ojeytonwilliams Could you clarify?

File: curriculum/challenges/english/03-front-end-libraries/react/render-state-in-the-user-interface.english.md [hunk 1]
```diff
@@ -31,7 +31,7 @@ tests:
...
     testString: assert(/<div><h1>.*<\/h1><\/div>/.test(Enzyme.mount(React.createElement(MyComponent)).html()));
   - text: The rendered <code>h1</code> header should contain text rendered from the component&apos;s state.
-    testString: 'async () => { const waitForIt = (fn) => new Promise((resolve, reject) => setTimeout(() => resolve(fn()), 250)); const mockedComponent = Enzyme.mount(React.createElement(MyComponent)); const first = () => { mockedComponent.setState({ name: ''TestName'' });   return waitForIt(() => mockedComponent.html()) }; const firstValue = await first(); assert(firstValue === ''<div><h1>TestName</h1></div>'');};'
+    testString: 'async () => { const waitForIt = (fn) => new Promise((resolve, reject) => setTimeout(() => resolve(fn()), 250)); const mockedComponent = Enzyme.mount(React.createElement(MyComponent)); const first = () => { mockedComponent.setState({ name: ''TestName'' });   return waitForIt(() => mockedComponent.html()) }; const firstValue = await first(); const getValue = firstValue.replace(' ', ''); assert(getValue === ''<div><h1>TestName</h1></div>'');};'
```
[the remark this answers is not in the document]
Comment (CheongJY) [reply curriculum/challenges/english/03-front-end-libraries/react/render-state-in-the-user-interface.english.md:34]: thanks for the update

**Comment by moT01** [comment]
The linting issue has to do with...
Screen Shot 2020-08-31 at 2 12 56 PM
The two single quotes are for escaping what is in them I believe. But in this case, we want to use them as single quotes in the replace method. When I remove `const getValue = firstValue.replace(' ', '');` - it passes. Trying to figure out how to escape those so the method recognizes them as the value.

**Comment by Sky020** [comment]
> Trying to figure out how to escape those so the method recognizes them as the value.

We could just wrap the test in double quotes, and use single quotes for in-between.

**Comment by RandellDawson** [comment]
Or just:
```yml
    testString: |
      async () => {
        const waitForIt = (fn) => new Promise((resolve, reject) => setTimeout(() => resolve(fn()), 250));
        const mockedComponent = Enzyme.mount(React.createElement(MyComponent));
        const first = () => {
          mockedComponent.setState({
            name: ''
            TestName ''
          });
          return waitForIt(() => mockedComponent.html())
       };
       const firstValue = await first();
       const getValue = firstValue.replace(' ', '');
       assert(getValue === '<div><h1>TestName</h1></div>');
      };
```

**Comment by moT01** [comment]
I have added @Sky020's suggested change and a fix to the linting issue.

**Review by RandellDawson — APPROVED** [review]
LGTM.  @Sky020 If you are ok with the changes @moT01 committed, feel free to merge this PR.

**Comment by Sky020** [comment]
Congrats on your first pull request (PR), @CheongJY ! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**